composer/models/base_hparams.py

```diff
@@ -1,19 +1,20 @@
 # Copyright 2021 MosaicML. All Rights Reserved.
 
-"""General `YAHP <https://docs.mosaicml.com/projects/yahp/en/stable/README.html>`_ interface for ComposerModels."""
+"""General `YAHP <https://docs.mosaicml.com/projects/yahp/en/stable/README.html>`_ interface for Base ComposerModels."""
```

Comment:
Would like to keep this file as `model_hparams.py` to match the style of `callback_hparams.py`, `algorithm_hparams.py`, ...

Reply:
Just checked `algorithm_hparams.py`. seems like algorithms hparams are all defined in their own top level `hparam.py` file and they all gain some loading functionality from the base `AlgorithmHparams`object.

That's not at all how things have been handled in the models module. Each model has its own hparams file. and though those hparams objects do inherit from `ModelHparams`.. they aren't gaining much from it. The only shared interfaces we're presupposing are `initializers` and `num_classes` and those are not universal to all models. 

Collecting all the model hparams in a shared file would make our codebase a lot cleaner but it's a odds with the pending `benchmarks` folder setup.